server/models/Document.ts

```diff
@@ -555,15 +561,19 @@ class Document extends ParanoidModel {
     return this.save(options);
   };
 
-  publish = async (userId: string, { transaction }: SaveOptions<Document>) => {
+  publish = async (
+    userId: string,
+    collectionId: string,
```

Comment:
Hope this makes more sense semantically from a functional perspective; a document is _supposed to be published in a collection_.

Reply:
Kind of, but I think you've now introduced a bug where you can publish into a different collection that the `document.collectionId` contains?

You need to check that `document.collectionId` is set inside the publish method or set it to the collectionId passed as an argument